pulp_ansible/app/models.py

```diff
@@ -227,6 +226,21 @@ class Meta:
         default_related_name = "%(app_label)s_%(model_name)s"
 
 
+class AnsibleCollectionDeprecated(Content):
```

Comment:
Can we generalize this? I'd rather not have to create a new content type for each piece of metadata that gets added down the line.

Reply:
Like we talked at our meetup about 1.5 weeks ago, we can do this but overall we think now isn't the right time. We could switch to a single model to store all the metadata (the rpm plugin has a few "bag" style models) at any point even if we continue with this PR as is (which is our proposal).

The reasoning is that while yes we will need additional models for additional metadata, in practice even if you put all the data into a generic, single model you'll still need additional code in various places anyway. So the simplicity it presents isn't really that beneficial in terms of avoiding shipping code to handle new metadata.

* it's unavoidable to have to parse the metadata for copying purposes. Copying a collection from one repo to another needs to bring some subset of the metadata. At that point having it unparsed to start with makes it harder.
* the data likely needs to be parsed to be included in the Galaxy APIs anyway. For example, with `deprecated` we'd have to parse this data anyway to be included in the Collection Serializer
* Merging remote metadata from a new sync, on-top of existing content which also has metadata requires joining. We can't just use data brought in at sync-time over metadata for collections already in the repo version.

So for now we're planning to keep this implementation pretty much as it is now, and we'll gladly generalize it later if that is helpful and when there are more examples for us to generalize with.

Reply:
@bmbouter I hope this doesn't lead us to a nasty migration later.

Reply:
@newswangerd I think it'll be straightforward given that the former objects are Content and the latter objects are Content.

Did you have any thoughts on the reasoning and conclusion we came to?

Reply:
@bmbouter I'm not terribly happy about it, but do what you must.

Reply:
I'm trying to explain that it's not a subjective design decision. There isn't an opportunity to make a "simple catch all solution" due to the bullets above that @daviddavis  and @fao89  and I could see. I can understand disappointment with the reality of that.

I'm unpacking the reasoning in bullets because I want this to be a shared decision. Can we reframe how this decision is being thought of to a "we must do what we must"? Basically I want your input to help us understand what we could do better, and if we can't find something, we accept the path we take together.

Reply:
@bmbouter I understand that this is our best option and I see why it is. I just wish there was a simpler solution for supporting more metadata in the future. I don't have any suggestions to improve it. You can go ahead and implement the proposal as you've outlined here.